### PR TITLE
feat: Add new timegrains and convert_dttm to Druid engine spec

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Add missing intermediate time grains and `convert_dttm` method to Druid engine spec.

### TEST PLAN
CI + new tests + tested locally

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/druid.py [hunk 1]
```diff
@@ -77,3 +82,12 @@ def get_extra_params(database: "Database") -> Dict[str, Any]:
             engine_params["connect_args"] = connect_args
             extra["engine_params"] = engine_params
         return extra
+
+    @classmethod
+    def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
```
Comment (ktmud) [inline superset/db_engine_specs/druid.py:87]: Make this `Enum`?

Comment (villebro) [reply superset/db_engine_specs/druid.py:87]: Good idea

File: superset/db_engine_specs/druid.py [hunk 2]
```diff
@@ -41,6 +42,10 @@ class DruidEngineSpec(BaseEngineSpec):  # pylint: disable=abstract-method
         None: "{col}",
         "PT1S": "FLOOR({col} TO SECOND)",
         "PT1M": "FLOOR({col} TO MINUTE)",
+        "PT5M": "TIME_FLOOR({col}, 'PT5M')",
+        "PT10M": "TIME_FLOOR({col}, 'PT10M')",
+        "PT15M": "TIME_FLOOR({col}, 'PT15M')",
+        "PT0.5H": "TIME_FLOOR({col}, 'PT30M')",
```
Comment (villebro) [inline superset/db_engine_specs/druid.py:48]: Added the intermediate grains between 1M and 1H.

**Review by ktmud — APPROVED** [review]
LGTM!

**Review by dpgaspar — APPROVED** [review]

**Pull request merged**